spec.html

```diff
@@ -122,6 +122,29 @@ <h1>
 		</emu-note>
 	</emu-clause>
 
+	<emu-clause id="sec-create-type-error-copy" type="abstract operation">
+		<h1>
+			CreateTypeErrorCopy (
+				_realmRecord_: a Realm Record,
+				_originalError_: an ECMAScript language value
+			)
+		</h1>
+		<emu-alg>
+			1. Let _newError_ be a newly created *TypeError* object.
+			1. NOTE: _newError_ is created in _realmRecord_.
+			1. If Type(_originalError_) is Object, then
+				1. Let _message_ be Completion(Get(_originalError_, "message")).
+				1. If _message_ is not an abrupt completion, then
+					1. If Type(_message_) is String, then
```

Comment:
this should use ToString - `message` is almost never not a string, and when it is, users will expect that coercion since everything in the language does it.

Reply:
I agree with you that `message` is almost never not a string, so we're talking about a low-priority edge case in terms of the developer impact. In general, when crossing a ShadowRealm boundary, we don't do coercions, we just throw when things don't work out. A coercion creates extra complexity, e.g., in error recovery. This is why I proposed the semantics that Caridy used, in a comment above. I don't feel extremely strongly about this, though; I'd be OK if this switched back to the previously proposed semantics.

Reply:
tbh that seems weirder than calling Get and just allowing any value (thus objects would throw) as a data property

Reply:
We need to close on both this question Jordan asks, as well as the suggestion from @mhofman that we avoid a Get and use a GetOwnProperty instead. I'm happy with this current spec text though :)

Reply:
I'm fine with this... although I noticed that firefox is doing something a little bit more elegant, they make the error message a composed one, e.g.:

```
// original message:
TypeError: null has no properties

// new message after crossing a boundary: 
Uncaught TypeError: wrapped function threw, error was TypeError: null has no properties
```

Which is a lot more descriptive. The stack is still needed, but the error is very good, at least during my tests, it was great that they have that level of detail. Now, I'm not sure how to spec that part... because of the nature of the message, which needs to be localized as well. I suspect we can just add a note allowing them to augment the message value. Or just remove the whole copy of the message spec, and just add a note for that allowing them to do this while keeping the spec simple. What do you think?

Reply:
As for calling Get, this is the conclusion that I have arrived to:

Accessing the `name` and `message` of the original object is observable, but considering that the error is crossing (been copied), meaning there is at least another point in the stack frame were some other code is interacting with the wrapped function, and that code can have a try/catch, and access the `name` and `message`, there is no way to make a clear distinction if these two properties were accessed because of the callable boundary, or was just the code on the caller controlling the error propagation.

Reply:
If we allow an augmented message, I guess we'd do that by not specifying the message at all, but rather explaining this possibility in the separate markdown file. I have no strong opinion whether we do this or the `Get` here, as long as we make a decision and document it.